File: clients/rust/src/api/cache.rs
// this file is @generated
use crate::{Configuration, error::Result, models::*};

pub struct Cache<'a> {
    cfg: &'a Configuration,
}

impl<'a> Cache<'a> {
    pub(super) fn new(cfg: &'a Configuration) -> Self {
        Self { cfg }
    }

    /// Cache Set
    pub async fn set(&self, cache_set_in: CacheSetIn) -> Result<CacheSetOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/cache/set")
            .with_body_param(cache_set_in)
            .execute(self.cfg)
            .await
    }

    /// Cache Get
    pub async fn get(&self, cache_get_in: CacheGetIn) -> Result<CacheGetOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/cache/get")
            .with_body_param(cache_get_in)
            .execute(self.cfg)
            .await
    }

    /// Get cache group
    pub async fn get_group(&self, cache_get_group_in: CacheGetGroupIn) -> Result<CacheGetGroupOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/cache/get-group")
            .with_body_param(cache_get_group_in)
            .execute(self.cfg)
            .await
    }

    /// Cache Delete
    pub async fn delete(&self, cache_delete_in: CacheDeleteIn) -> Result<CacheDeleteOut> {
        crate::request::Request::new(http::Method::POST, "/api/v1/cache/delete")
            .with_body_param(cache_delete_in)
            .execute(self.cfg)
            .await
    }
}

File: z-clients/javascript/src/apis/cache.ts
// this file is @generated

import {
    type CacheDeleteIn,
    CacheDeleteInSerializer,
} from '../models/cacheDeleteIn';
import {
    type CacheDeleteOut,
    CacheDeleteOutSerializer,
} from '../models/cacheDeleteOut';
import {
    type CacheGetIn,
    CacheGetInSerializer,
} from '../models/cacheGetIn';
import {
    type CacheGetOut,
    CacheGetOutSerializer,
} from '../models/cacheGetOut';
import {
    type CacheSetIn,
    CacheSetInSerializer,
} from '../models/cacheSetIn';
import {
    type CacheSetOut,
    CacheSetOutSerializer,
} from '../models/cacheSetOut';
import { CacheNamespace } from './cacheNamespace';
import { HttpMethod, DiomRequest, type DiomRequestContext } from "../request";

export class Cache {
    public constructor(private readonly requestCtx: DiomRequestContext) {}

    public get namespace() {
        return new CacheNamespace(this.requestCtx);
    }

    /** Cache Set */
    public set(
        key: string,
        cacheSetIn: CacheSetIn,
    ): Promise<CacheSetOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.cache.set");

        request.setBody(
            CacheSetInSerializer._toJsonObject({
                ...cacheSetIn,
                key: key,
            })
        );
        
        return request.send(
            this.requestCtx,
            CacheSetOutSerializer._fromJsonObject,
        );
    }/** Cache Get */
    public get(
        key: string,
        cacheGetIn: CacheGetIn,
    ): Promise<CacheGetOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.cache.get");

        request.setBody(
            CacheGetInSerializer._toJsonObject({
                ...cacheGetIn,
                key: key,
            })
        );
        
        return request.send(
            this.requestCtx,
            CacheGetOutSerializer._fromJsonObject,
        );
    }/** Cache Delete */
    public delete(
        key: string,
        cacheDeleteIn: CacheDeleteIn,
    ): Promise<CacheDeleteOut> {
        const request = new DiomRequest(HttpMethod.POST, "/api/v1.cache.delete");

        request.setBody(
            CacheDeleteInSerializer._toJsonObject({
                ...cacheDeleteIn,
                key: key,
            })
        );
        
        return request.send(
            this.requestCtx,
            CacheDeleteOutSerializer._fromJsonObject,
        );
    }
}

